app/services/policies/policies_list_service.rb

```diff
@@ -10,26 +10,49 @@ class Policies::PoliciesListService
 
   class PoliciesListServiceError < RuntimeError; end
 
-  def self.call(account, builtin: true)
+  def self.apicast_registry_url
+    ThreeScale.config.sandbox_proxy.apicast_registry_url
+  end
+
+  delegate :apicast_registry_url, to: 'self.class'
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(account, builtin: true)
+    @account = account
+    @builtin = builtin
+  end
+
+  attr_reader :account
+
+  def builtin?
+    !!@builtin
+  end
+
+  def call
     list = PolicyList.new
-    list.merge! fetch_policies_from_apicast if builtin
-    list.merge! policies_from_account(account)
+    list.merge! fetch_policies_from_apicast if builtin?
+    list.merge! policies_from_account
     list.to_h
   rescue *HTTP_ERRORS, PoliciesListServiceError => error
     Rails.logger.error { error } and return
   end
 
-  def self.fetch_policies_from_apicast
+  private
+
+  def fetch_policies_from_apicast
     response = ::JSONClient.get(apicast_registry_url)
     raise_policies_list_error(response.content) unless response.ok?
     response.body['policies']
   end
 
-  def self.raise_policies_list_error(error)
+  def raise_policies_list_error(error)
     raise PoliciesListServiceError, I18n.t('errors.messages.apicast_not_found', url: apicast_registry_url, error: error)
   end
 
-  def self.policies_from_account(account)
+  def policies_from_account
     return unless account.provider_can_use?(:policy_registry)
     PolicyList.new(account.policies)
```

Comment:
No need for `@account` here?

Reply:
@josemigallas `account` is calling the method `account`, which exists for the `attr_reader :account` which is the same as:
```ruby
def account
  @account
end
```

Reply:
Nope. It's here already: https://github.com/3scale/porta/blob/5496b80bdec77587ae325fff4dd3591fc706e0f9/app/services/policies/policies_list_service.rb#L28